src/squlearn/qnn/lowlevel_qnn_pennylane.py

```diff
@@ -0,0 +1,980 @@
+
+    Args:
+        key (Union[str, tuple, ParameterVector, ParameterVectorElement]): Key of the derivative for
+                                                                          the value dictionary
+        order (int): Order of the derivative
+        argnum (Union[None, List[int]]): List of which arguments (x,param,param_obs) to
+                                         differentiate
+        return_grad_param (bool): Calculate the gradient with respect to the parameters
+        return_grad_param_obs (bool): Calculate the gradient with respect to the observable
+                                      parameters
+        return_grad_x (bool): Calculate the gradient with respect to the input
+        squared (bool): Calculate the squared value of the derivative
+    """
+
+    def __init__(
+        self,
+        key: Union[str, tuple, ParameterVector, ParameterVectorElement],
+        order: int = 0,
+        argnum: Union[None, List[int]] = None,
+        return_grad_param: bool = False,
+        return_grad_param_obs: bool = False,
+        return_grad_x: bool = False,
+        squared: bool = False,
+    ):
+        self.key = key
+        self.order = order
+        self.argnum = argnum
+        self.return_grad_param = return_grad_param
+        self.return_grad_param_obs = return_grad_param_obs
+        self.return_grad_x = return_grad_x
+        self.squared = squared
+
+
+class post_processing_evaluation:
+    """Container class for the post processing evaluation of derivatives of the QNN
+
+    Args:
+        key (Union[str, tuple, ParameterVector, ParameterVectorElement]): Key of the derivative
+                                                                          for the value dictionary
+        evaluation_tuple (tuple): Tuple of direct_evaluation objects that are used for evaluating
+                                  the values that are used in the post processing
+        evaluation_function (callable): Function to evaluate the post processing
+    """
+
+    def __init__(
+        self,
+        key: Union[str, tuple, ParameterVector, ParameterVectorElement],
+        evaluation_tuple: tuple,
+        evaluation_function: callable,
+    ):
+        self.key = key
+        self.evaluation_tuple = evaluation_tuple
+        self.evaluation_function = evaluation_function
+
+
+def _get_class_from_string(
+    val: Union[
+        str,
+        tuple,
+        direct_evaluation,
+        post_processing_evaluation,
+        ParameterVector,
+        ParameterVectorElement,
+    ]
+) -> Union[direct_evaluation, post_processing_evaluation]:
+    """Converts an input string to the direct or post processing evaluation object
+
+    Args:
+        val (Union[str, tuple, direct_evaluation, post_processing_evaluation,
+                ParameterVector, ParameterVectorElement]): Input string or evaluation object
+
+    Returns:
+        Associated Expec object
+
+    """
+
+    def eval_var(value_dict: dict) -> np.ndarray:
```

Comment:
why are these defs indented? Not a fan

Reply:
definetly needs to be changed, since this function is called everytime evaluate is called

Reply:
Done